### PR TITLE
Project Management Part 2

### PR DESCRIPTION
**Pull request by abbiehuynh**
## PR includes:
**Implementing useContext for managing project states**
- Improves separation of concern by creating a separate useContext for managing projects states
  - updateProjects function was previously in Auth Context Provider, which manages the authentication state for tokens, login/logout functions.
  - Project Context now includes: fetchProjects(), deleteProject(), updateProjects().

## Changes:
- There are no UI Changes, and app should work and look the same as it did in the previous PR

### Notes:
- This is part 2 of project management. I wanted to implement useContext and creating a separate provider before continuing to add more routes/endpoints for projects 
- **Currently, the function to add a new project by completing a project name form is located in AddProjectModal.jsx. Should this function be added to the Project Context, or remain in this component?**
- Please review the app structure and suggest any improvements for the logic of applying useContext to manage project state. 
- The only commit to review is [1c1c961](https://github.com/abbiehuynh/my-crochet-corner/pull/6/commits/1c1c961984e8fab46814c580235659ad7f5f8d03) (please ignore other commits)

---
Merges project-management branch into projects branch (part 2)

### CODE REVIEW

**Comment by davidmedina-contact** [comment]
> Currently, the function to add a new project by completing a project name form is located in AddProjectModal.jsx. Should this function be added to the Project Context, or remain in this component?

In general it is better to separate API functions from the UI component

File: client/src/components/home/ProjectProvider.jsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        } 
+    };
+
+    useEffect(() => {
+        // fetches projects only if userId is available - will not try to fetch before user has logged in
+        if (userId) {
+            fetchProjects();
+        } else {
+            // clear projects if user is logged out
+            setProjects([]);
```
Comment (davidmedina-contact) [inline client/src/components/home/ProjectProvider.jsx:79]: Is there more that needs to be cleared or done during logout?

**Review by davidmedina-contact — APPROVED** [review]

File: client/src/components/home/ProjectProvider.jsx (hunk 1 above)
Comment (abbiehuynh) [reply client/src/components/home/ProjectProvider.jsx:79]: In relation to projects, I do not think so. There is more done during the logout process in the AuthProvider Component, such as clearing the token and authentication/access to the app

**Comment by mai-repo** [comment]
Hi **@abbiehuynh**,

Overall, this is a solid PR. Implementing useContext for managing project states is a great idea, and I appreciate your approach to separating concerns by centralizing the API calls. My only suggestion is to add more error handling on the front end for users. Everything else looks good! There are no excessive console.log statements, and your comments for all functions and components enhance the readability of your code.

**Comment by abbiehuynh** [comment]
> Hi **@abbiehuynh**,
> 
> Overall, this is a solid PR. Implementing useContext for managing project states is a great idea, and I appreciate your approach to separating concerns by centralizing the API calls. My only suggestion is to add more error handling on the front end for users. Everything else looks good! There are no excessive console.log statements, and your comments for all functions and components enhance the readability of your code.

@Maithanhceu Thank you so much for your review, Mai! I appreciate your feedback, and I will take your suggestion for adding more error handling into consideration.

**Pull request merged**